libs/ui-toolkit/src/components/toast/use-toasts.ts

```diff
@@ -0,0 +1,66 @@
+import create from 'zustand';
+import type { Toast } from './toast';
+
+type ToastsStore = {
+  /**
+   * A list of active toasts
+   */
+  toasts: Toast[];
+  /**
+   * Adds/displays a new toast
+   */
+  add: (toast: Toast) => void;
+  /**
+   * Updates a toast
+   */
+  update: (id: string, newToast: Toast) => void;
+  /**
+   * Closes a toast
+   */
+  close: (id: string) => void;
+  /**
+   * Closes all toasts
+   */
+  closeAll: () => void;
+  /**
+   * Arbitrary removes a toast
+   */
+  remove: (id: string) => void;
+  /**
+   * Arbitrary removes all toasts
+   */
+  removeAll: () => void;
+};
+
+export const useToasts = create<ToastsStore>((set) => ({
+  toasts: [],
+  add: (toast) =>
+    set((state) => ({
+      toasts: [...state.toasts, toast],
+    })),
+  update: (id, newToast) =>
+    set((state) => {
+      const toasts = [...state.toasts];
+      const toastIdx = toasts.findIndex((t) => t.id === id);
+      if (toastIdx > -1) toasts[toastIdx] = newToast;
+      return { toasts };
+    }),
+  close: (id) =>
```

Comment:
question (non blocking): Do we want to remove that toast from the toasts array completely. I guess this is problematic with the animation. Perhaps we should reach for react-spring or similar to make this easier to handle.

Reply:
I'd leave it as is and change it later once we'll have the animation guideline ready.